### PR TITLE
code cleanup: remove vestiges of baggage header, and fix delta xDS for WDS

### PR DESCRIPTION
**Pull request by kyessenov**

### CODE REVIEW

**Comment by zirain** [comment]
Funny: a google guy fails google CLA, lol.

**Comment by kyessenov** [comment]
@zirain it's non-blocking. We stopped using `baggage` in istio (search for baggage), so it's safe to remove the unused code.

**Comment by zirain** [comment]
looks like Google CLA down at that time, will review this on Monday.

File: source/extensions/filters/http/peer_metadata/config.proto [hunk 1]
```diff
@@ -20,6 +20,7 @@ package io.istio.http.peer_metadata;
 // Peer metadata provider filter. This filter encapsulates the discovery of the
 // peer telemetry attributes for consumption by the telemetry filters.
 message Config {
+  // DEPRECATED.
```
Comment (zirain) [inline source/extensions/filters/http/peer_metadata/config.proto:23]: when will this be removed? next major release?

**Review by zirain — APPROVED** [review]
LGTM, add `DNM` for @GregHanson take a review.

**Review by GregHanson — APPROVED** [review]
lgtm

**Pull request merged**